raft/src/test/java/org/apache/kafka/raft/internals/VoterSetHistoryTest.java

```diff
@@ -91,11 +92,40 @@ void testAddAt() {
         assertEquals(Optional.of(removedVoterSet), votersHistory.valueAtOrBefore(200));
     }
 
+    @Test
+    void testAddAtNonOverlapping() {
+        Map<Integer, VoterSet.VoterNode> voterMap = VoterSetTest.voterMap(IntStream.of(1, 2, 3), true);
+        VoterSet staticVoterSet = VoterSet.fromMap(new HashMap<>(voterMap));
+        VoterSetHistory votersHistory = new VoterSetHistory(staticVoterSet, new LogContext());
+
+        // Assert multiple voters can be added or removed at a time
+        voterMap.put(4, VoterSetTest.voterNode(4, true));
+        voterMap.put(5, VoterSetTest.voterNode(5, true));
+
+        VoterSet addedMultipleVoters = VoterSet.fromMap(new HashMap<>(voterMap));
+        votersHistory.addAt(100, addedMultipleVoters);
```

Comment:
This doesn't do what your comment indicates. Before this PR, voter set were allowed to have more than one change if the last voter set was the static voter set.

See the `testAddAtNonOverlapping` method that this PR removes to see how to construct this case correctly.

Let's also move this method to be below the `testBootstrapAddAt` method so that git and Github can generate a better diff.